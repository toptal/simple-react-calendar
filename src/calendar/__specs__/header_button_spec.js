import React from 'react'

import HeaderButton from '../header_button'
import ReactTestRenderer from 'react-test-renderer'

describe('HeaderButton', () => {
  const defaultProps = {
    type: 'prev',
    enabled: true,
    onClick: () => {},
  }

  it('renders <HeaderButton />', () => {
    const tree = ReactTestRenderer.create(<HeaderButton {...defaultProps} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  context('when prop `enabled` is `false`', () => {
    it('renders <HeaderButton />', () => {
      const modProps = Object.assign(defaultProps, {enabled: false})
      const tree = ReactTestRenderer.create(<HeaderButton {...modProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  context('when prop `type` is "next"', () => {
    it('renders <HeaderButton />', () => {
      const modProps = Object.assign(defaultProps, {type: 'next'})
      const tree = ReactTestRenderer.create(<HeaderButton {...modProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  context('when prop `blockClassName` is defined', () => {
    it('renders <HeaderButton />', () => {
      const modProps = Object.assign(defaultProps, {blockClassName: 'cal'})
      const tree = ReactTestRenderer.create(<HeaderButton {...modProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  context('when prop `title` is defined', () => {
    it('renders <HeaderButton />', () => {
      const modProps = Object.assign(defaultProps, {title: 'testTitle'})
      const tree = ReactTestRenderer.create(<HeaderButton {...modProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  context('when prop `arrow` is defined', () => {
    it('renders <HeaderButton />', () => {
      const modProps = Object.assign(defaultProps, {arrow: <i className="icon">Test</i>})
      const tree = ReactTestRenderer.create(<HeaderButton {...modProps} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
