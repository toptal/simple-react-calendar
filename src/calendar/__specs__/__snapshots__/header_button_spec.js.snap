// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev"
  disabled={false}
  onClick={[Function]}
  title={undefined}
  type="button"
/>
`

exports[`HeaderButton when prop \`arrow\` is defined renders <HeaderButton /> 1`] = `
<button
  className="cal-header_button is-next is-disabled"
  disabled={true}
  onClick={[Function]}
  title="testTitle"
  type="button"
>
  <i
    className="icon"
  >
    Test
  </i>
</button>
`

exports[`HeaderButton when prop \`blockClassName\` is defined renders <HeaderButton /> 1`] = `
<button
  className="cal-header_button is-next is-disabled"
  disabled={true}
  onClick={[Function]}
  title={undefined}
  type="button"
/>
`

exports[`HeaderButton when prop \`enabled\` is \`false\` renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev is-disabled"
  disabled={true}
  onClick={[Function]}
  title={undefined}
  type="button"
/>
`

exports[`HeaderButton when prop \`title\` is defined renders <HeaderButton /> 1`] = `
<button
  className="cal-header_button is-next is-disabled"
  disabled={true}
  onClick={[Function]}
  title="testTitle"
  type="button"
/>
`

exports[`HeaderButton when prop \`type\` is "next" renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-next is-disabled"
  disabled={true}
  onClick={[Function]}
  title={undefined}
  type="button"
/>
`
